packages/modules/region/src/models/country.ts

```diff
@@ -1,44 +1,28 @@
-import { Entity, ManyToOne, PrimaryKey, Property } from "@mikro-orm/core"
-
-import { createPsqlIndexStatementHelper } from "@medusajs/utils"
+import { model } from "@medusajs/utils"
 import Region from "./region"
 
-// We don't need a partial index on deleted_at here since we don't support soft deletes on countries
-const regionIdIsoIndexName = "IDX_region_country_region_id_iso_2_unique"
-const RegionIdIsoIndexStatement = createPsqlIndexStatementHelper({
-  name: regionIdIsoIndexName,
-  tableName: "region_country",
-  columns: ["region_id", "iso_2"],
-  unique: true,
-})
-
-RegionIdIsoIndexStatement.MikroORMIndex()
-@Entity({ tableName: "region_country" })
-export default class Country {
-  @PrimaryKey({ columnType: "text" })
-  @Property({ columnType: "text" })
-  iso_2: string
-
-  @Property({ columnType: "text" })
-  iso_3: string
-
-  @Property({ columnType: "text" })
-  num_code: string
-
-  @Property({ columnType: "text" })
-  name: string
-
-  @Property({ columnType: "text" })
-  display_name: string
-
-  @Property({ columnType: "text", nullable: true })
-  region_id?: string | null = null
-
-  @ManyToOne({
-    entity: () => Region,
-    fieldName: "region_id",
-    nullable: true,
-    onDelete: "set null",
-  })
-  region?: Region | null
-}
+const Country = model
+  .define(
+    { name: "Country", tableName: "region_country" },
+    {
+      iso_2: model.text().primaryKey(),
+      iso_3: model.text(),
+      num_code: model.text(),
+      name: model.text(),
```

Comment:
Should the country code and name be searchable as well?

Reply:
Yeah, I guess it doesn't hurt. Will add that in